### PR TITLE
Add Derivative Aggregation

### PR DESCRIPTION
**Pull request by maltehuebner**
I added a class to perform `Derivative` aggregations and wrote a small test case for it.

### CODE REVIEW

File: lib/Elastica/Aggregation/Derivative.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    /**
+     * Set the gap policy for this aggregation.
+     *
+     * @param string $gapPolicy
+     *
+     * @return $this
+     */
+    public function setGapPolicy($gapPolicy)
```
[the remark this answers is not in the document]
Comment (p365labs) [reply lib/Elastica/Aggregation/Derivative.php]: @ruflin I suspected that :) but not sure tnx again.

**Comment by maltehuebner** [comment]
I added your feedback to the pull request, please have a look on it :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -52,6 +52,22 @@ All notable changes to this project will be documented in this file based on the
       "number_of_significant_value_digits" : 3
     }
   ```
+
+### Added
+
+* Added a transport class for mocking a HTTP 403 error codes, useful for testing response failures in inheriting clients
```
Comment (ruflin) [inline CHANGELOG.md]: @maltehuebner You don't seem to have luck with the changelog and rebase :-(

Comment (maltehuebner) [reply CHANGELOG.md]: Next try. I’m sure I’ll never get a job again if someone sees my git merging skills.

**Comment by p365labs** [comment]
@ruflin for me it's ok to be merged

**Comment by ruflin** [comment]
@p365labs Go for it ;-)

File: CHANGELOG.md (hunk 2 above)
Comment (ruflin) [reply CHANGELOG.md]: No worries, it's not you. It's the 3-way merge commit magic from Git that soemtimes does funny stuff here :-)

**Comment by ruflin** [comment]
@p365labs Yes

**Pull request merged**